### ISSUE TITLE
hubble does not set drop reason description field for L7 drop events

### ISSUE THREAD

**Issue by michi-covalent** [cilium/cilium#22402]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

hubble always sets `DropReasonDesc` field to `DROP_REASON_UNKNOWN` for L7 drop events:

- https://github.com/cilium/cilium/blob/master/pkg/hubble/parser/seven/parser.go#L126

in `hubble_drop_total` metric, these L7 drop events get `reason="Success"` label, which is sort of misleading: 

- https://github.com/cilium/cilium/blob/master/pkg/hubble/metrics/drop/handler.go#L56
- https://github.com/cilium/cilium/blob/master/pkg/monitor/api/drop.go#L18

ideally these L7 drop events should contain a more descriptive drop reason like `POLICY_DENIED`.

### Cilium Version

`v1.12.4`

### Kernel Version

```
Linux minikube 5.15.64-0-virt #1-Alpine SMP Mon, 05 Sep 2022 08:02:49 +0000 aarch64 aarch64 aarch64 GNU/Linux
```

### Kubernetes Version

```
Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.3", GitCommit:"816c97ab8cff8a1c72eccca1026f7820e93e0d25", GitTreeState:"clean", BuildDate:"2022-01-25T21:25:17Z", GoVersion:"go1.17.6", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.3", GitCommit:"816c97ab8cff8a1c72eccca1026f7820e93e0d25", GitTreeState:"clean", BuildDate:"2022-01-25T21:19:12Z", GoVersion:"go1.17.6", Compiler:"gc", Platform:"linux/arm64"}
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by dylandreimerink** [comment]
@michi-covalent can you give a quick update. I see we have some PRs merged for this issue, others closed due to inactivity. Can you describe what work is still left to do?

**Comment by michi-covalent** [comment]
yeah we still need something like https://github.com/cilium/cilium/pull/22409. it became stale while waiting for some confirmation from @jrajahalme. currently i don't have cycles to work on this, so i unassigned myself.

**Comment by kaworu** [comment]
@michi-covalent can you take a look please? 🙏

**Comment by michi-covalent** [comment]
yeah looks like it's been fixed. let's close it
